core/server/worker/src/main/java/alluxio/worker/block/DefaultWorkerMetaStore.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
+
+import static alluxio.Constants.CLUSTERID_FILE;
+
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.util.io.FileUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+/**
+ * persistence worker info to a file.
+ */
+@NotThreadSafe
+public class DefaultWorkerMetaStore implements WorkerMetaStore {
```

Comment:
We don't need an interface and default implementation. You can make `WorkerMetaStore` the one class that has the logic. If the Factory is not necessary, we can remove that too.

I admit that in our existing code, we have a lot of interfaces + 1 default implementation. That's not too necessary and we don't have to follow that pattern. But I appreciate the try.